problem-spring-web-autoconfigure/src/main/java/org/zalando/problem/spring/web/autoconfigure/JacksonModulesRegistratorWithoutMvcAutoConfig.java

```diff
@@ -0,0 +1,35 @@
+package org.zalando.problem.spring.web.autoconfigure;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.zalando.problem.ProblemModule;
+import org.zalando.problem.violations.ConstraintViolationProblemModule;
+
+import java.util.List;
+
+/**
+ * Registers jackson modules when using @EnableWebMvc, which deactivates WebMvc AutoConfig
+ */
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnBean(WebMvcConfigurationSupport.class)
+@AutoConfigureBefore(WebMvcAutoConfiguration.class)
+public class JacksonModulesRegistratorWithoutMvcAutoConfig implements WebMvcConfigurer {
```

Comment:
This class has no conflict resolution in case someone uses `@EnableWebMvc` and configures a custom `MappingJackson2HttpMessageConverter` by hand already. From experience that's a pretty common occurrence. I'd argue that if someone chooses `@EnableWebMvc` then everything should be configured by hand.

Reply:
>     2\. my expectation when using `@EnableWebMvc` would be to affect the behaviour of spring itself, not a third-party library. Say, I enable it for some other reason specific to spring, then i will break problem-spring-web and will have to come here to find how to manually configure it, which seems cumbersome

I can follow that line of though.